src/unix/process.c

```diff
@@ -424,20 +426,162 @@ int uv_spawn(uv_loop_t* loop,
 
   /* Acquire write lock to prevent opening new fds in worker threads */
   uv_rwlock_wrlock(&loop->cloexec_lock);
-  pid = fork();
 
-  if (pid == -1) {
-    err = UV__ERR(errno);
-    uv_rwlock_wrunlock(&loop->cloexec_lock);
-    uv__close(signal_pipe[0]);
-    uv__close(signal_pipe[1]);
-    goto error;
-  }
+/* Especial child process spawn case for macOS Big Sur (11.0) onwards 
+ *
+ * Big Sur introduced a significant performance degradation on a call to
+ * fork/exec when the process has many pages mmaped in with MAP_JIT, like, say
+ * a javascript interpreter. Electron-based applications, for example,
+ * are impacted; though the magnitude of the impact depends on how much the
+ * app relies on subprocesses. 
+ * 
+ * On macOS, though, posix_spawn is implemented in a way that does not 
+ * exhibit the problem. This block implements the forking and preparation
+ * logic with poxis_spawn and its related primitves. It also takes advantage of 
+ * the macOS extension POSIX_SPAWN_CLOEXEC_DEFAULT that makes impossible to
+ * leak descriptors to the child process.
+ * 
+ * see https://github.com/libuv/libuv/issues/3050
+ */
+#if defined(__APPLE__)
+  if (__builtin_available(macOS 10.15, *)) {
+    int fd;
+    sigset_t signal_set;
+    posix_spawnattr_t attrs;
+    posix_spawn_file_actions_t actions;
+
+    err = posix_spawnattr_init(&attrs);
+    if(err)
```

Comment:
```suggestion
    if (err != 0)
```
(I won't point out more individual instances but please fix them all.)